packages/editor/src/package/storage-format.ts

```diff
@@ -0,0 +1,129 @@
+import { EditorPluginType } from '@editor/types/editor-plugin-type'
+import { TemplatePluginType } from '@editor/types/template-plugin-type'
+import * as t from 'io-ts'
+import { v4 as uuid_v4 } from 'uuid'
+
+import { getEditorVersion } from './editor-version'
+
+/** The creator of the saved data -> Serlo editor */
+const documentType = 'https://serlo.org/editor'
+
+type Migration = (state: { version: number }) => { version: number }
+
+const migrations: Migration[] = [
+  // Migration: Add `editorVersion` and `id`
+  (state): StorageFormat => {
+    const expectedType = t.type({
+      type: t.literal(documentType),
+      variant: EditorVariantType,
+      version: t.number,
+      dateModified: t.string,
+      document: t.type({
+        plugin: t.string,
+        state: t.unknown,
+      }),
+    })
+    if (!expectedType.is(state))
+      throw new Error(
+        `Unexpected type during migration. Expected ${JSON.stringify(expectedType)} but got ${JSON.stringify(state)}`
+      )
+
+    const editorVersion = getEditorVersion()
+    const id = uuid_v4()
+
+    return {
+      ...state,
+      editorVersion,
+      id,
+    }
+  },
+  // ...
+  // Add new migrations here. Make sure the last one returns the new StorageFormat.
+]
+
+const currentVersion = migrations.length
+
+export function createEmptyDocument(
+  editorVariant: EditorVariant
+): StorageFormat {
+  return {
+    id: uuid_v4(),
+    type: documentType,
+    variant: editorVariant,
+    version: currentVersion,
+    editorVersion: getEditorVersion(),
+    dateModified: getCurrentDatetime(),
+    document: {
+      plugin: TemplatePluginType.GenericContent,
+      state: {
+        content: {
+          plugin: EditorPluginType.Rows,
+        },
+      },
+    },
+  }
+}
```

Comment:
Would you consider extracting this function out of this file, like you did `getEditorVersion`?

Reply:
Hm, I tried to move it but it depends on `documentType` & `currentVersion` which only make sense in context of `storage-format`. I think I would leave it there for now if that is fine for you?